BudgetTracker.Business/BudgetPeriods/BudgetPeriod.cs

```diff
@@ -1,13 +1,19 @@
+using BudgetTracker.Business.Budgeting;
 using System;
 
-namespace BudgetTracker.Common.Models
+namespace BudgetTracker.Business.BudgetPeriods
 {
-    public class BudgetPeriod 
+    public class BudgetPeriod
     {
         /// <summary>
-        /// The budget id related to this budget period.
+        /// The root budget related to this budget period.
         /// </summary>
-        public Guid BudgetId { get; set; }
+        public Budget RootBudget { get; set; }
+
+        /// <summary>
+        /// The root budget id related to this budget period.
+        /// </summary>
+        public Guid RootBudgetId { get; set; }
```

Comment:
Why do we need the `RootBudgetId` if we already have the RootBudget? and if we don't want the `RootBudget` then why don't we just get it using any of the get budget metohds.

Reply:
So that you can still have that link without loading the actual Budget from the repo. This is especially important when the user sends a request and we build this from the request. The user only has to specify the budget by ID, they don't have to pass in the whole budget object.

Reply:
That makes sense, but why does the transaction need to save the whole budget itself.

Reply:
what do you mean by "save the whole budget itself"?

Reply:
Oh this is the business layer, so we aren't actually saving the RootBudgetId and the Budget to the data base via entity framework.